libraries/AP_Scripting/examples/command_int.lua

```diff
@@ -0,0 +1,45 @@
+--[[
+   demonstrate using the gcs:command_int() interface to send commands from MAVLink MAV_CMD_xxx set
+--]]
+
+local MAV_FRAME_GLOBAL = 0
+
+local MAV_CMD_DO_SET_MODE = 176
+local MAV_CMD_DO_REPOSITION = 192
+
+-- some plane flight modes for testing
+local MODE_LOITER = 12
+local MODE_GUIDED = 15
+
+local MAV_MODE_FLAG_CUSTOM_MODE_ENABLED = 1
+
+--[[
+   test API calls. When in LOITER mode change to GUIDED and force flying to a location NE of home
+--]]
+local function test_command_int()
+   if vehicle:get_mode() ~= MODE_LOITER then
+      return
+   end
+   local home = ahrs:get_home()
+   if not home then
+      return
+   end
+
+   -- force mode GUIDED
+   gcs:command_int(MAV_FRAME_GLOBAL, MAV_CMD_DO_SET_MODE, { MAV_MODE_FLAG_CUSTOM_MODE_ENABLED, MODE_GUIDED })
```

Comment:
The vehicle:update_target_location(old_location, new_location) method validates the old location hasn't changed before setting new location to "prevent a race condition where the user changes mode or commands a different target in the controlling lua script". Is that not a problem when using MAVLink?